src/Pages/ADT3DScenePage/ADT3DScenePage.tsx

```diff
@@ -231,6 +238,34 @@ const ADT3DScenePageBase: React.FC<IADT3DScenePageProps> = ({
         [deeplinkDispatch, environmentPickerOptions?.adt]
     );
 
+    // if the adx connection information is not in the environment context. fetch it and update the context
+    useEffect(() => {
+        if (!adapter.getADXConnectionInformation()) {
+            {
+                connectionState.callAdapter({
+                    adtInstanceIdentifier: deeplinkState.adtResourceId
+                        ? { id: deeplinkState.adtResourceId }
+                        : { hostName: deeplinkState.adtUrl?.slice(7) }
+                });
+                dispatch({
```

Comment:
it's usually a code smell when you need 2 dispatch operations in a row. Is there a way we can combine them into a single action so that it's more of an "action" rather than just a buch of "setState" calls?